packages/integration-karma/test/wire/wiring/index.spec.js

```diff
@@ -70,7 +70,22 @@ describe('wiring', () => {
             expect(updateCalls[2].method).toBe('update'); // overriddenInChild
         });
 
-        it('should be called next tick when component with wire that has dynamic params is created', () => {
+        it('should be called after connect when ENABLE_WIRE_DEFERRED_FIRST_UPDATE=false (default) and component with wire that has dynamic params is created', () => {
+            const spy = [];
+            AdapterId.setSpy(spy);
+            expect(spy.length).toBe(0);
+
+            const elm = createElement('x-echo-adapter-consumer', { is: ComponentClass });
+            document.body.appendChild(elm);
+
+            expect(spy).toHaveSize(2);
+            expect(spy[0].method).toBe('connect');
+            expect(spy[1].method).toBe('update');
+        });
+
+        it('should be called next tick when ENABLE_WIRE_DEFERRED_FIRST_UPDATE=true and component with wire that has dynamic params is created', () => {
+            setFeatureFlagForTest('ENABLE_WIRE_DEFERRED_FIRST_UPDATE', true);
```

Comment:
If the test fails for any reason, the flag is never reset to its original value. I would rather create a new `describe` block with `beforeEach` and `afterEach` block to conditionally enable this flag.